### ISSUE TITLE
Support should_panic for rune tests

### ISSUE THREAD

**Issue by Roba1993** [rune-rs/rune#653]
Allow should panic flag for rune test functions like this:
```rust
#[test]
#[should_panic]
pub fn test() {
    let object = #{ field: 1 };
    let object2 = object;
    println!("field: {}", object.field);
    // move the variable into the drop function and consume it
    drop(object2);
    // This will fail
    println!("field: {}", object.field);
}
```

**Comment by udoprog** [comment]
This is already supported for doc tests btw.

**Comment by Roba1993** [comment]
> This is already supported for doc tests btw.

Yeah I have seen that. My problem is for our documentation with Zola (non API documentation) I load scripts from a file. This allows me to test them and use them in the documentation in parallel.
See an example here: [Exolynk Documentation](https://dev.exolynk.app/docs/docs/script/control-flow/)

For that I need to explain the should_panic within the rune script itself.
